staging/src/k8s.io/kubectl/pkg/metricsutil/metrics_printer.go

```diff
@@ -200,49 +202,70 @@ func (printer *TopCmdPrinter) PrintPodMetrics(metrics []metricsapi.PodMetrics, p
 		if withNamespace {
 			printValue(w, NamespaceColumn)
 		}
+		if len(pods.Items) > 0 {
+			printValue(w, HostColumn)
+			printValue(w, IpColumn)
+		}
 		if printContainers {
 			printValue(w, PodColumn)
 		}
 		printColumnNames(w, PodColumns)
 	}
 
 	sort.Sort(NewPodMetricsSorter(metrics, withNamespace, sortBy))
-
+	podMap := createPodMap(pods)
 	for _, m := range metrics {
+		pod, ok := podMap[m.Name]
 		if printContainers {
 			sort.Sort(NewContainerMetricsSorter(m.Containers, sortBy))
-			printSinglePodContainerMetrics(w, &m, withNamespace)
+			printSinglePodContainerMetrics(w, &m, withNamespace, pod, ok)
 		} else {
-			printSinglePodMetrics(w, &m, withNamespace)
+			printSinglePodMetrics(w, &m, withNamespace, pod, ok)
 		}
 	}
 	return nil
 }
 
+func createPodMap(pods *v1.PodList) (podMap map[string]v1.Pod) {
+	podMap = make(map[string]v1.Pod)
+	for _, pod := range pods.Items {
+		podMap[pod.Name] = pod
+	}
+	return podMap
+}
+
 func printColumnNames(out io.Writer, names []string) {
 	for _, name := range names {
 		printValue(out, name)
 	}
 	fmt.Fprint(out, "\n")
 }
 
-func printSinglePodMetrics(out io.Writer, m *metricsapi.PodMetrics, withNamespace bool) {
+func printSinglePodMetrics(out io.Writer, m *metricsapi.PodMetrics, withNamespace bool, pod v1.Pod, podExists bool) {
 	podMetrics := getPodMetrics(m)
 	if withNamespace {
 		printValue(out, m.Namespace)
 	}
+	if podExists {
```

Comment:
hum, can't this be something like "if pod != nil"? Instead of passing the value and the boolean?
Example:
```golang
        podMap := make(map[string]interface{})
	podMap["a"] = "aaa"
	podMap["b"] = "bbb"
	v, _ := podMap["c"]
	if v != nil {
		fmt.Println(v) // Will print nothing
	}
        x, _ := podMap["a"]
	if x != nil {
		fmt.Println(x) // Will print "aaa"
	}
```

Reply:
Yeah was not 100% sure of the best way to determine if a pod exists in the map. But it looks like the zero value of a interface is nil, so this works.

Reply:
So actually doing it like this breaks. Since this is mapping to a struct Pod type. the zero value is not nil and this does not work. Its throwing an error. do you have another suggested method or should we continue to use the boolean returned from the map when a key does not exist?

The purpose of this podMap is to make sure we can quickly lookup pods and print out the information relative the pod without having to loop through the PodList every time.